lib/puppet/transaction.rb

```diff
@@ -374,6 +402,9 @@ def skip?(resource)
       unless resource.class == Puppet::Type.type(:whit) then
         resource.warning _("Skipping because of failed dependencies")
       end
+    elsif resource_status(resource).failed? and
+      @prefetch_failed_providers[resource.type][resource.provider.class.name]
+      resource.debug "Skipping because failed" #Do not try to evaluate a resource with a known failed provider
```

Comment:
skipping what because what failed?

Can you give someone an type of hint of what failed? Or where they should be looking?

Reply:
So, a few things about the message here:

1. The message here should be a warning, not a debug. A resource being skipped because the provider prefetch failed is at least as severe as a resource being skipped because its dependencies failed.
2. Should (as Eric pointed out) be expanded to say "Skipping because provider prefetch failed".
3. Should be wrapped in `_( ... )` so it'll get translated.

Having "Skipped because provider prefetch failed" should be enough in the message, since additional information gets added by the warning method so it ends up looking something like this:

```
Warning: /Stage[main]/Main/File[/tmp/bar]: Skipped because provider prefetch failed
```